composer/loggers/object_store_logger.py

```diff
@@ -0,0 +1,383 @@
+
+            The following format variables are available:
+
+            +------------------------+-------------------------------------------------------+
+            | Variable               | Description                                           |
+            +========================+=======================================================+
+            | ``{artifact_name}``    | The name of the artifact being logged.                |
+            +------------------------+-------------------------------------------------------+
+            | ``{run_name}``         | The name of the training run. See                     |
+            |                        | :attr:`~composer.core.logging.Logger.run_name`.       |
+            +------------------------+-------------------------------------------------------+
+            | ``{rank}``             | The global rank, as returned by                       |
+            |                        | :func:`~composer.utils.dist.get_global_rank`.         |
+            +------------------------+-------------------------------------------------------+
+            | ``{local_rank}``       | The local rank of the process, as returned by         |
+            |                        | :func:`~composer.utils.dist.get_local_rank`.          |
+            +------------------------+-------------------------------------------------------+
+            | ``{world_size}``       | The world size, as returned by                        |
+            |                        | :func:`~composer.utils.dist.get_world_size`.          |
+            +------------------------+-------------------------------------------------------+
+            | ``{local_world_size}`` | The local world size, as returned by                  |
+            |                        | :func:`~composer.utils.dist.get_local_world_size`.    |
+            +------------------------+-------------------------------------------------------+
+            | ``{node_rank}``        | The node rank, as returned by                         |
+            |                        | :func:`~composer.utils.dist.get_node_rank`.           |
+            +------------------------+-------------------------------------------------------+
+
+            Leading slashes (``'/'``) will be stripped.
+
+            Consider the following example, which subfolders the artifacts by their rank:
+
+            >>> object_store_logger = ObjectStoreLogger(..., object_name_format='rank_{rank}/{artifact_name}')
+            >>> trainer = Trainer(..., run_name='foo', logger_destinations=[object_store_logger])
+            >>> trainer.logger.file_artifact(..., artifact_name='bar.txt', file_path='path/to/file.txt')
+
+            Assuming that the process's rank is ``0``, the object store would store the contents of
+            ``'path/to/file.txt'`` in an object named ``'rank_0/bar.txt'``.
+
+            Default: ``'{artifact_name}'``
```

Comment:
Ahhh, yes.

I originally had it this way, but in PR 8 (I know, sorry), I had the caller include the run name by default. This is so that artifacts will be subfoldered by run on the local disk. If the object store logger appends the run name (again) here, then it will be double-subfoldered in the object store.